ch_tools/common/logging.py

```diff
@@ -0,0 +1,189 @@
+"""
+Logging module.
+"""
+
+import inspect
+import logging
+from typing import Any, Dict, Optional
+
+from loguru import logger
+
+this: Dict[str, Any] = {}
+
+
+class Filter:
+    """
+    Filter for luguru handler.
+    """
+
+    def __init__(self, name):
+        self._name = name
+
+    def __call__(self, record):
+        """
+        Filter callback to decide for each logged message whether it should be sent to the sink or not.
+        """
+
+        return record["extra"].get("logger_name") == self._name
+
+
+def make_filter(name):
+    """
+    Factory for filter creation.
+    """
+
+    return Filter(name)
+
+
+class InterceptHandler(logging.Handler):
+    """
+    Helper class for logging interception.
+    """
+
+    def emit(self, record: logging.LogRecord) -> None:
+        """
+        Intercept all records from the logging module and redirect them into loguru.
+
+        The handler for loguru will be chosen based on module name.
+        """
+
+        # Get corresponding Loguru level if it exists.
+        level: int or str  # type: ignore
+        try:
+            level = logger.level(record.levelname).name
+        except ValueError:
+            level = record.levelno
+
+        # Find caller from where originated the logged message.
+        frame, depth = inspect.currentframe(), 0
+        while frame and (depth == 0 or frame.f_code.co_filename == logging.__file__):
+            frame = frame.f_back
+            depth += 1
+
+        logger.bind(logger_name=record.name).opt(
+            depth=depth, exception=record.exc_info
+        ).log(level, record.getMessage())
+
+
+def configure(config_loguru: dict, module: str, cmd_name: Optional[str] = None) -> None:
```

Comment:
```suggestion
def configure(config_loguru: dict, module: str, extra_opt: dict: ) -> None:
```
IMO it is not necessary to use `cmd name`  as a parameter. Lets make it scalable.

Reply:
There are two kinds of parameters which we can add to logger. `cmd_name` is used in `bind()` so it will be added to `extra` dictionary and can be used in formatting. There are also `opt()` args, such as `exception`, there are no much usage for others, but I added both `extra` and `opt` arguments for `configure` just in case.